src/System.Text.Json.Dynamic/System/Text/Json/JsonDynamicObject.cs

```diff
@@ -10,7 +10,17 @@ namespace System.Text.Json
 {
     public class JsonDynamicObject : DynamicObject, IBufferFormattable
     {
-        Dictionary<JsonProperty, JsonValue> _properties;
+        #region Constants
+
+        private static readonly char[] EncodingChars = new char[] { '{', '}', ',', ':', '"' };
+        private static readonly ReadOnlySpan<char> OpenCurlySpan = new ReadOnlySpan<char>(EncodingChars, 0, 1);
+        private static readonly ReadOnlySpan<char> CloseCurlySpan = new ReadOnlySpan<char>(EncodingChars, 1, 1);
+        private static readonly ReadOnlySpan<char> CommaSpan = new ReadOnlySpan<char>(EncodingChars, 2, 1);
+        private static readonly ReadOnlySpan<char> ColonSpan = new ReadOnlySpan<char>(EncodingChars, 3, 1);
```

Comment:
I think this is fine for now, but once the compiler checks are implemented, you won't be able to have span fields in classes. At that point, this will become expensive. 

I think it might be worth adding single character to to code-point encode methods to the encoders. What we don't want is single character to a buffer.